### ISSUE TITLE
Standarize convert NodeList to Array

### ISSUE THREAD

**Issue by boton** [liferay/liferay-frontend-guidelines#51]
Array has a lot of helpful methods, for working a NodeList first, you need to convert in Array.

We have some options:
- `Array.prototype.slice.call(nodeList)`
- `Array.from(nodeList)`
- `[...nodeList]`

Working in a `<aui:script` trying to remove some old AUI, I need to transform nodeList in Array and I use the first option because is compatible out of the box with ie11, but at the end `Array.from(nodeList)` is covered by our polyfill.

Because of this, I think that `Array.from(nodeList)` works great in all our cases transpiled/untranspiled code.

**Comment by wincent** [comment]
Yes, and `Array.from` is polyfilled for IE in liferay-portal, and the polyfill works with `NodeList`.

In any "*.es.js" file you can safely use `[...nodeList]`.

So the choice is between a simple recommendation (eg. "use `Array.from` everywhere") and a nuanced one (eg. "use `Array.from` in untranspiled code, and `[...spread]` in transpiled code). I think simple guidelines (easily understood, easily applied, uniform) are usually best, but at the same time I am hoping that our transpilation coverage will be going up a lot this year as we move more code into React, so we could also consider "skating towards where the puck is going" and optimizing our recommendations for that (hopefully imminent future) scenario.

**Comment by wincent** [comment]
Just to update this, since the last comment, we are effectively transpiling *all* JS (apart from a very tiny bit in themes) which means we are free to use spread syntax if we want.

Having said that, I don't think I would recommend spread syntax from this use case.

- `Array.from(thing)`: Makes it clear that my intention is to convert a potentially non-`Array` thing to an `Array`; on the other hand:
-  `[...thing]`: The intent is ambiguous, because I may be wanting to convert an iterable to an `Array` *or* I may be wanting to make a copy of something I don't own, so that I can mutate it.

So if we wanted to close this I think we'd just need to turn it into a document in this repo (I doubt this is something we'd make a lint for).
